pkg/staticnetworkconfig/generator.go

```diff
@@ -308,6 +309,67 @@ func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistence(macInterf
 	return nil
 }
 
+func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistenceYAML(macInterfaceMap models.MacInterfaceMap, networksYaml string) error {
+	interfaceNames := lo.Map(macInterfaceMap, func(m *models.MacInterfaceMapItems0, _ int) string { return m.LogicalNicName })
+
+	var config map[string]interface{}
+
+	// Unmarshal the JSON string into the config struct
+	err := yaml.Unmarshal([]byte(networksYaml), &config)
+	if err != nil {
+		s.log.WithError(err).Errorf("Error unmarshalling JSON")
+		return err
+	}
+
+	// See 'man systemd.net-naming-scheme' for interface naming protocol
+	predicatableIfaceNamePattern := regexp.MustCompile("^en[PsvxXbucaipod]")
```

Comment:
If we really must have two separate functions doing the validation, at least make this global so it can't get out of sync.

Reply:
As I agree with you, so I'll stick with the validation on YAML, so no need to make it global